File: src/frontend/src/services/transactionService.ts
import api from './api';

const transactionService = {
  getTransactions: async (params: any) => {
    const response = await api.get('/transactions', { params });
    return response.data;
  },

  createTransaction: async (data: any) => {
    const response = await api.post('/transactions', data);
    return response.data;
  },

  updateTransaction: async (id: number, data: any) => {
    const response = await api.put(`/transactions/${id}`, data);
    return response.data;
  },

  deleteTransaction: async (id: number) => {
    const response = await api.delete(`/transactions/${id}`);
    return response.data;
  }
};

export default transactionService;

File: src/backend/src/main/java/com/personal/finance/backend/transactions/service/TransactionService.java
package com.personal.finance.backend.transactions.service;

import com.personal.finance.backend.transactions.dto.request.CreateTransactionRequest;
import com.personal.finance.backend.transactions.dto.request.UpdateTransactionRequest;
import com.personal.finance.backend.transactions.dto.response.TransactionDTO;
import com.personal.finance.backend.transactions.dto.response.TransactionHistoryDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;
import java.util.List;

public interface TransactionService {
    TransactionDTO createTransaction(Long userId, CreateTransactionRequest request);
    Page<TransactionDTO> filterTransactions(Long userId, Long walletId, Long categoryId, LocalDate startDate, LocalDate endDate, String keyword, Pageable pageable);
    void deleteTransaction(Long id, Long userId);
    TransactionDTO updateTransaction(Long id, Long userId, UpdateTransactionRequest request);
    List<TransactionHistoryDTO> getTransactionHistory(Long id, Long userId);
}